csharp/extractor/Semmle.Extraction.CSharp/Kinds/TypeAnnotation.cs

```diff
@@ -0,0 +1,13 @@
+﻿namespace Semmle.Extraction.Kinds
+{
+    enum TypeAnnotation
+    {
+        NotApplicable,
+        Disabled,
```

Comment:
I think there's a more fundamental problem, which is `TypeAnnotation` should be `[Flags]` since the options are not, in general, mutually exclusive. Then there's no need for `TypeAnnotation?` to indicate `NotApplicable`/`Disabled` and we can indeed remove them.